exporter/collector/googlemanagedprometheus/extra_metrics_test.go

```diff
@@ -63,9 +66,103 @@ func TestAddTargetInfo(t *testing.T) {
+				sm.Scope().SetName("myscope")
+				sm.Scope().SetVersion("v0.0.1")
+
+				// other metrics should not be copied to target_info
+				metric := sm.Metrics().AppendEmpty()
+				metric.SetName("baz-metric")
+				metric.SetEmptyGauge().DataPoints().AppendEmpty().SetIntValue(2112)
+				return metrics
+			}(),
+			expected: func() pmetric.ResourceMetricsSlice {
+				rms := pmetric.NewResourceMetricsSlice()
+				rm := rms.AppendEmpty()
+
+				// resource attributes will be changed to a MonitoredResource on export, not in AddTargetInfo
+				// therefore they should still be present here
+				rm.Resource().Attributes().PutStr(locationLabel, "us-east")
+				rm.Resource().Attributes().PutStr("foo-label", "bar")
+
+				scopeInfo := rm.ScopeMetrics().AppendEmpty()
+				scopeInfoMetric := scopeInfo.Metrics().AppendEmpty()
+				scopeInfoMetric.SetName("otel_scope_info")
+				scopeInfoMetric.SetEmptyGauge().DataPoints().AppendEmpty().SetIntValue(1)
+				scopeInfoMetric.Gauge().DataPoints().At(0).Attributes().PutStr("otel_scope_name", "myscope")
+				scopeInfoMetric.Gauge().DataPoints().At(0).Attributes().PutStr("otel_scope_version", "v0.0.1")
+				return rms
+			}(),
+		},
+		{
+			name: "add both scope info and target info",
+			testFunc: func(m pmetric.Metrics) pmetric.ResourceMetricsSlice {
+				extraMetrics := AddTargetInfoMetric(m)
+				AddScopeInfoMetric(m).MoveAndAppendTo(extraMetrics)
+				return extraMetrics
+			},
+			input: func() pmetric.Metrics {
+				metrics := pmetric.NewMetrics()
+				rm := metrics.ResourceMetrics().AppendEmpty()
+
+				// foo-label should be copied to target_info, not locationLabel
+				rm.Resource().Attributes().PutStr(locationLabel, "us-east")
+				rm.Resource().Attributes().PutStr("foo-label", "bar")
+
+				// scope should not be copied to target_info
+				sm := rm.ScopeMetrics().AppendEmpty()
+				sm.Scope().SetName("myscope")
+				sm.Scope().SetVersion("v0.0.1")
+
+				// other metrics should not be copied to target_info
+				metric := sm.Metrics().AppendEmpty()
+				metric.SetName("baz-metric")
+				metric.SetEmptyGauge().DataPoints().AppendEmpty().SetIntValue(2112)
+				return metrics
+			}(),
+			expected: func() pmetric.ResourceMetricsSlice {
+				rms := pmetric.NewResourceMetricsSlice()
+				rm := rms.AppendEmpty()
+
+				// resource attributes will be changed to a MonitoredResource on export, not in AddTargetInfo
+				// therefore they should still be present here
+				rm.Resource().Attributes().PutStr(locationLabel, "us-east")
+				rm.Resource().Attributes().PutStr("foo-label", "bar")
+
+				sm := rm.ScopeMetrics().AppendEmpty()
+				metric := sm.Metrics().AppendEmpty()
+				metric.SetName("target_info")
+				metric.SetEmptyGauge().DataPoints().AppendEmpty().SetIntValue(1)
+				metric.Gauge().DataPoints().At(0).Attributes().PutStr("foo-label", "bar")
+
+				rm2 := rms.AppendEmpty()
+				rm2.Resource().Attributes().PutStr(locationLabel, "us-east")
+				rm2.Resource().Attributes().PutStr("foo-label", "bar")
```

Comment:
note that the way I'm doing this, the resources get copied over (we're basically building 2 separate metricSlices and appending them to the main slice). is this ok?

Reply:
Of course we rather would not, but if we have to it is fine.  What would the alternative be?  Make the function return a ScopeMetricSlice instead of a ResourceMetricSlice?

Reply:
Yeah, I was trying to keep both functions with the same signature as the ExtraMetrics extension point for future flexibility but I guess it's not really necessary

Reply:
couldn't target info also be returned as a ScopeMetricSlice?

Reply:
We need a ScopeMetricSlice for each Resource though right? How about the changes in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/571/commits/6be421c043c1b9acfe9cc5cfdaa31b676fac582a#diff-0af4ea6286cefe60de85c9cad15924d571a1755ae70b74638c2b36355c46554aR108-R111